definitions/tools/gather_to_sub_directory_dirs.cwl

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: ["/bin/bash","directory_gatherer.sh"]
+
+requirements:
+    - class: DockerRequirement
+      dockerPull: "ubuntu:xenial"
+    - class: ResourceRequirement
+      ramMin: 1000
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'directory_gatherer.sh'
+        entry: |
+            set -eou pipefail
+
+            outdir="$1"
+            files="${@:2}"
+            mkdir $outdir
+            chmod -R 777 $outdir
+            cp --recursive --preserve --no-clobber --target-directory $outdir $files
```

Comment:
I'll advocate for quoting the parameters here, too.

Reply:
Quoting the `files` variable during the `cp` command has the command think it's a single file with spaces instead of multiple space separated files. This causes the tool to fail unless there is a single input file. I kept thinking it was a compute cache layer issue. Have any other suggestions?

Reply:
The trick here is to keep the `$files` variable as a bash array, too.

So something like:
```bash
files=("${@:2}")
...
cp ... "${files[@]}"
```

Reply:
TIL! That method worked, thanks.